sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -217,11 +220,11 @@ static const alert_info_t kExpectedInfo[kRoundTotal] = {
         {
             .test_name = "All classes",
             .alert_cause =
-                ((uint64_t)1 << kTopEarlgreyAlertIdFlashCtrlRecovErr) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdKmacFatalFaultErr) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdKeymgrRecovOperationErr) |
-                ((uint64_t)1 << kTopEarlgreyAlertIdKeymgrFatalFaultErr),
-            .class_accum_cnt = {0, 1, 0, 0x1e02},
+                ((uint64_t)1 << kTopEarlgreyAlertIdRvCoreIbexRecovSwErr) |
+                ((uint64_t)1 << kTopEarlgreyAlertIdUart0FatalFault) |
+                ((uint64_t)1 << kTopEarlgreyAlertIdI2c0FatalFault) |
+                ((uint64_t)1 << kTopEarlgreyAlertIdSpiHost0FatalFault),
+            .class_accum_cnt = {1, 1, 1, 1},
             .class_esc_state = {kCstateIdle, kCstatePhase1, kCstateIdle,
```

Comment:
.class_esc_state = {kCstateIdle, kCstatePhase1, kCstateIdle, kCstatePhase0},

the values posted here are minimum value. (see line 570)
So if captured phase is greater than equal to these, check will pass.

I had to do this way, because phase capture is done in current or next phase
when we escalate alert in multiple phases.
Not sure that is the intent but at least that is what I see in the waveform.
(@msfschaffner correct me if this is not the case)

Given that,
kCstateIdle means check will pass if "captured phase" >= idle, which basically means "blind pass".
Since we are expecting classA,B,C and D in test round 3,
I would prefer
 .class_esc_state =  {kCstatePhase0, kCstatePhase1, kCstatePhase0, kCstatePhase0},
If fpga can't capture uart and spi  alert, we can put the same if esle statement like line 472.

Reply:
let me change this back to that and see what happened. I think @alphan might have changed this before because there was a minor bug in the FPGA and because the ping timeout was kind of screwing up the test flow.

It might work as expected now.

Reply:
fixed this!